CounterApp/Components/UserInfo/UserInfoView.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  UserInfo.swift
+//  CounterApp
+//
+//  Created by Wenjuan Li on 2022/7/22.
+//
+
+import ComposableArchitecture
+import SwiftUI
+
+struct UserInfoView: View {
+    let store: Store<UserInfoState, UserInfoAction>
+    
+    var body: some View {
+        WithViewStore(self.store.scope(state: \.viewState)) { viewStore in
+            VStack {
+                HStack {
+                    Text("Frist Name: ").padding(.trailing)
+                    TextField(
+                        "",
+                        text: viewStore.binding(get: \.firstName, send: UserInfoAction.firstNameChanged)
+                    )
+                }
+                
+                HStack {
+                    Text("Last Name: ").padding(.trailing)
+                    TextField(
+                        "",
+                        text: viewStore.binding(get: \.lastName, send: UserInfoAction.lastNameChanged)
+                    )
+                }
+                
+                HStack {
+                    Text("Email: ").padding(.trailing)
+                    TextField(
+                        "",
+                        text: viewStore.binding(get: \.email, send: UserInfoAction.emailChanged)
+                    ).keyboardType(.emailAddress)
+                }
+                
+                HStack {
+                    Text("Age: ").padding(.trailing)
+                    TextField(
+                        "",
+                        text: viewStore.binding(get: \.age, send: UserInfoAction.ageChanged)
+                    ).keyboardType(.numberPad)
+                }
+                
+                HStack {
+                    Text("Job: ").padding(.trailing)
+                    TextField(
+                        "",
+                        text: viewStore.binding(get: \.job, send: UserInfoAction.jobChanged)
+                    )
+                }
+            }.disabled(viewStore.disabled)
```

Comment:
To make this view more composable and also being able to remove the `disabled` from state, you could remove this view modifier from here and just let the callers decide whether they want this view disabled or not i.e. `UserDetailView` will apply this view modifier while `EditUserView` will not.

Reply:
@nanocba Thanks! Fixed!